### DISCUSSION TITLE
KAIN 200 AIMO

### DISCUSSION THREAD

**Discussion by CanedAndUnable** [X3n0m0rph59/eruption#72] [Hardware Support]
Hi, would it be possible to add a battery level for this mouse like in windows?  I'm pretty new to Linux and have been searching everywhere for support for the roccat mouse and keyboard and your led solution has been the first one to work for me.  Just gotta say excellent work and thanks ^^

**Comment by X3n0m0rph59** [comment]
Hi @CanedAndUnable, reporting of the current battery status is best implemented at the level of the Linux kernel and/or the desktop. The Eruption daemon currently does not query the battery levels of a device.

Do you see any indication of your device when you execute the following command?
```shell
$ upower --dump
```

If you are unsure, you can post the output of the aforementioned command as a follow-up to this message.

**Comment by X3n0m0rph59** [comment]
Hi @CanedAndUnable, Eruption version `0.1.23` most likely will ship with support for showing battery levels and signal strength indicators of `ROCCAT Kain 200/202` series wireless mice. The respective values are read directly from the device, therefore this feature does not require a kernel patch or something like that. I have implemented this feature in the daemon proper, the yet-to-be-released tech demo of the `Eruption GUI` and in the [`Eruption Profile Switcher`](https://extensions.gnome.org/extension/2621/eruption-profile-switcher/) GNOME Shell Extension since `v18`.

![screenshot-profile-switcher-01](https://user-images.githubusercontent.com/3053519/134301840-8a01c56a-b08c-44b4-9a85-6dc6104aa2c6.jpg)

![screenshot-profile-switcher-02](https://user-images.githubusercontent.com/3053519/134301842-eba48437-bfb8-49ea-8f33-737fac2642d6.jpg)


![screenshot-01](https://user-images.githubusercontent.com/3053519/134301259-70094019-dca8-4d35-8c71-bb819ca610d5.png)
